src/orgwarden/repo_crawler.py

```diff
@@ -17,50 +18,81 @@ def __init__(self, hostname: str, message: str):
 
 
 def fetch_org_repos(
-    org_name: str,
-    hostname: str,
+    org_name: str, hostname: str, gh_pat: str | None = None
 ) -> list[Repository]:
     """
     Returns a `Repository` list containing the specified organization's public, non-forked repositories.
-    Raises an `APIError` if an error occurs while fetching repositories.
-    Raises a `FileNotFoundError` if the GitHub CLI is not installed.
+    Raises an `AuthError` if OrgWarden lacks authorization with GitHub API.
+    Raises an `APIError` if an other error occurs while fetching repositories, or the JSON response does not match expected schema.
     """
 
     if not org_name:
         raise ValueError("org_name is an empty string.")
     if not hostname:
         raise ValueError("hostname is an empty string.")
 
-    # check if gh cli is installed
-    if shutil.which("gh") is None:
-        raise FileNotFoundError("GitHub CLI is not installed.")
+    BASE_URL = (
+        "https://api.github.com"
+        if "github.com" in hostname or "www.github.com" in hostname
+        else f"https://{hostname}/api/v3"
+    )
+    headers = {
+        "Accept": "application/vnd.github+json",
+        "X-GitHub-Api-Version": "2022-11-28",
+    }
+    if not gh_pat:
+        try:  # Load PAT to avoid rate-limits in local testing
+            import dotenv
```

Comment:
The logic to check the environment if a token isn't provided doesn't seem right here, especially given that the intent is for it to be used in local testing scenarios. In fact, this will break as written if someone uses OrgWarden as a dependency because `python-dotenv` is only installed as a dev dependency.

Move this functionality to the test itself.